types/underscore/index.d.ts

```diff
@@ -8,8 +8,9 @@
 //                 Florian Keller <https://github.com/ffflorian>
 //                 Regev Brody <https://github.com/regevbr>
 //                 Piotr Błażejewicz <https://github.com/peterblazejewicz>
+//                 Michael Ness <https://github.com/reubenrybnik>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
-// TypeScript Version: 2.8
+// TypeScript Version: 3.0
```

Comment:
This PR specifically doesn't need this version bump, but I'm planning on using `unknown` and generic tuple rest parameters in future PRs if I can. I could avoid doing that for now or wait until we actually get to July, though, if it would be preferable to do so.

Reply:
With regards to the use of `unknown`, I got that recommendation from [this possibly knockoff Do's and Dont's page](http://www.staging-typescript.org/docs/handbook/declaration-files/do-s-and-don-ts.html#any), though I'm really confused now about how I got there in the first place since that's not the page that the DefinitelyTyped readme links to. Thinking more about this, while I personally would prefer the use of `unknown` for projects I might use this definition in, it probably wouldn't provide a good experience to those trying to migrate from JS to TS since for a JS file with no type information at all the chain of everything being `any` would be broken at every Underscore function, so maybe I should change this back to using `any` instead of `unknown`.

With regards to the use of generic tuple rest parameters, re-looking through the full set of changes I'm not seeing any of the generic tuple rest parameters that I claimed aside from notes on updating zip and unzip once 3.1 is the minimum version, so if `any` is preferable to `unknown` I should probably revert this change.

Reply:
Two things.

- Regarding the TS version: please bump it in the first PR in which you actually start depending on a newer TS version.
- Regarding `unknown`: that knockoff doesn't remotely look like the [official do's and dont's page](https://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html). The latter doesn't say anything about `unknown`. It also appears that `unknown` is not just a type-safe alternative to `any`; it has [very different semantics](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-0.html#new-unknown-top-type). In any case, I don't think you should make a wholesale change replacing all occurrences of `any` by `unknown`, but given the different semantics, there might be particular instances of `any` that would be more appropriately typed `unknown`.

Reply:
It's true, I feel super silly for having ended up on that page at all. Sounds good, I think I'll just switch everything back to `any` then since that will allow me to leave the TS version unchanged 😄

Reply:
Hooray, I don't feel quite as silly now, it looks like that knockoff page with its new guidance on the use of `unknown` vs. `any` is likely coming to a typescriptlang.org near us very soon! Still not sure how I ended up there early though.

![image](https://user-images.githubusercontent.com/8692126/86537310-1cc76e00-bea3-11ea-8fdd-de9883b1b362.png)
![image](https://user-images.githubusercontent.com/8692126/86537320-29e45d00-bea3-11ea-8ff3-ca82badeee99.png)

Reply:
So you were apparently looking into the future.

They clearly have a strong preference for `unknown` over `any`, which is understandable, but I still think a wholesale replacement of `any` by `unknown` would be unwise given the huge semantic difference.

Reply:
I agree, I'd definitely try to use `unknown` when writing new type definitions from scratch but you're right that such a change would be likely to cause a lot of errors and resentment for folks moving between versions of existing definitions.